node/src/components/contract_runtime/operations.rs

```diff
@@ -161,6 +208,70 @@ pub fn execute_finalized_block(
                         .cloned()
                 },
             );
+
+    // Purging
+
+    if let Some(previous_block_height) = finalized_block.height().checked_sub(1) {
+        match calculate_prune_eras(
+            activation_point_era_id,
+            key_block_height_for_activation_point,
+            previous_block_height,
+            prune_batch_size,
+        ) {
+            Some(keys_to_prune) => {
+                let first_key = keys_to_prune.first().copied();
+                let last_key = keys_to_prune.last().copied();
+                info!(
+                    previous_block_height,
+                    %key_block_height_for_activation_point,
+                    %state_root_hash,
+                    first_key=?first_key,
+                    last_key=?last_key,
+                    "commit_prune: preparing purge config");
+                let prune_config = PruneConfig::new(state_root_hash, keys_to_prune);
+                match engine_state.commit_prune(CorrelationId::new(), prune_config) {
+                    Ok(PruneResult::RootNotFound) => {
+                        error!(
+                            previous_block_height,
+                            %state_root_hash,
+                            "commit_prune: root not found");
```

Comment:
shouldn't we kill the node in this case?

Reply:
I can do this. It's impossible to hit this case. CC @Fraser999